de-lib/src/main/java/org/iplantc/de/diskResource/client/views/search/cells/DiskResourceQueryForm.java

```diff
@@ -354,22 +75,34 @@ public void show(Element parent, AnchorAlignment anchorAlignment) {
         fireEvent(new ShowEvent());
     }
 
-    /*
-     * (non-Javadoc)
-     *
-     * @see com.sencha.gxt.widget.core.client.menu.Menu#onHide()
-     *
-     * When this container becomes hidden, ensure that save filter container is hidden as well.
-     *
-     * Additionally, this will perform any desired animations when this form is hidden.
-     */
     @Override
-    protected void onHide() {
-        namePrompt.hide();
-        super.onHide();
+    public void renderSearchForm(ReactSearchForm.SearchFormProps props) {
+        CyVerseReactComponents.render(ReactSearchForm.SearchForm,
+                                      props,
+                                      DivElement.as(con.getElement()));
+    }
+
+    @Override
+    public void hide() {
+        if (showing) {
+            onHide();
+            RootPanel.get().remove(this);
+            eventPreview.remove();
+            showing = false;
+            hidden = true;
```

Comment:
I might be missing something but I don't see `hidden` used anywhere.

Reply:
Ahh got it, it is hiding the component when set. But In that case can we check the if the component is visible just using `hidden` and get rid of `showing` ?